lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -255,6 +257,51 @@ getBlockedEncoding(tt::LoadOp loadOp, ModuleAxisInfoAnalysis &axisInfo) {
                                        threadsPerWarp, CTALayout);
 }
 
+static ttg::SharedEncodingAttr
+getSharedEncoding(tt::LoadOp loadOp, Operation *use, bool isMMAV3,
+                  ttg::DotOperandEncodingAttr dotOpEnc, bool needTrans) {
+
+  auto ty = loadOp.getType().cast<RankedTensorType>();
+  ttg::SharedEncodingAttr sharedEnc;
+  auto CTALayout = ttg::getCTALayout(ty.getEncoding());
+  auto blockedOrder = ttg::getOrder(ty.getEncoding());
+  SmallVector<unsigned> order;
+  if (blockedOrder.size() == 3) {
+    for (unsigned i = 0; i < blockedOrder.size(); ++i) {
+      if (blockedOrder[i] == 0)
+        continue;
+      order.push_back(blockedOrder[i]);
+    }
+    order.push_back(0);
+  } else {
+    order = blockedOrder;
+  }
+  if (isa<tt::DotOp>(use)) {
+    if (dotOpEnc) {
+      assert(!isMMAV3 && "MMAv3 should not have dotOperandEncoding");
+      unsigned bitWidth = ty.getElementType().getIntOrFloatBitWidth();
+      // set needTrans to avoid unnecessary conversion between shared encodings.
+      sharedEnc =
+          ttg::SharedEncodingAttr::get(ty.getContext(), dotOpEnc, ty.getShape(),
+                                       order, CTALayout, bitWidth, needTrans);
+    } else {
+      assert(isMMAV3 && "Load used by dot op should be either MMAv3 or have "
+                        "dotOperandEncoding.");
+      sharedEnc =
+          ttg::SharedEncodingAttr::get(ty.getContext(), ty.getShape(), order,
+                                       CTALayout, ty.getElementType());
+    }
+  } else {
+    assert(!dotOpEnc && !isMMAV3 &&
+           "Load used by non-dot op should not have dotOperandEncoding or be "
+           "MMAv3.");
+    // Use non-swizzled layout for loads that do not feed into dot ops.
+    sharedEnc = ttg::SharedEncodingAttr::get(ty.getContext(), 1, 1, 1, order,
+                                             CTALayout);
+  }
```

Comment:
I wonder if this is the right condition. We want to skip swizzling because the load is 1D and therefore we don't expect bank conflicts. Why the user being Dot or not matters?

Reply:
I was thinking about that too... Let me see if I can handle this on the `SharedEncodingAttr::get` to pick the layout right for given tensor

Reply:
that's a solution however there could be cases where someone wants to swizzle 1D tensors. I would just keep the logic in there. 

BTW:
```
sharedEnc = ttg::SharedEncodingAttr::get(ty.getContext(), 1, 1, 1, order,
                                             CTALayout);
```
Are the async.cp vectorized when using this layout?

Reply:
What would it mean to swizzle a 1D tensor?  I thought that was just the identity.

Reply:
If `row == 0`, then if I'm reading this math right, then `phase` is always 0, so the col index is unchanged?  https://github.com/openai/triton/blob/main/third_party/amd/lib/TritonAMDGPUToLLVM/TritonGPUToLLVMBase.h#L285-L289

Anyway if you're right then the comment added in this PR is definitely wrong.  :)

Reply:
I wonder about 1D swizzled tensor too. For now I have put the rank checking in here, locally. See what do you think.

Reply:
> For now I have put the rank checking in here, locally. See what do you think.

If we need the rank checking there (i.e. if rank-1 swizzle is not the identity), then the comment is wrong, right?  I feel like now is our opportunity to figure out what the heck is going on here, I'd hate to waste it.

Reply:
I added the support for 1D in [this](https://github.com/openai/triton/pull/2438) PR, and the changes are basically to not do swizzling on 1D.

Reply:
@pawelszczerbuk are you seeing swizzled addresses when you use the original shared layout for the 1D tensor (i.e. when vec != 1, max_phase != 1) ? is that what is causing the perf hit, or is it the extra unnecessary arithmetics that are calculating the "swizzled" addresses ?

Reply:
So, what I see is in case of vec=1, per_phase=1, max_phase=8 I see one cp.async of length 0x10, for vec=2, per_phaase=1, max_phase=8 I see two cp.asyncs of length 0x8. That explains the perf difference.